deb/openmediavault/usr/share/php/openmediavault/system/system.inc

```diff
@@ -28,6 +28,31 @@ require_once("openmediavault/functions.inc");
  * @ingroup api
  */
 class System {
+	/**
+	 * Get a partition parent block device
+	 * @return Returns the storage parent device file of the partiton of where the
+	 * operating system is installed
+	 * @throw \OMV\ExecException
+	 */
+	public static function getBlockParentDevice($blockdevices, $partiton) {
+		static $blockParentDevice = NULL;
```

Comment:
Why is this a static variable?

Reply:
Out put of` lsblk -J -o NAME` command
```
root@openmediavault:~# lsblk -J -o NAME
{
   "blockdevices": [
      {"name":"sda",
         "children": [
            {"name":"sda1"},
            {"name":"sda2"},
            {"name":"sda5",
               "children": [
                  {"name":"sda5_crypt",
                     "children": [
                        {"name":"nas--vg-root"},  <---- Here is my root mount point /
                        {"name":"nas--vg-swap_1"}
                     ]
                  }
               ]
            }
         ]
      },
      {"name":"sdb"},
      {"name":"sdc"},
      {"name":"sr0"}
   ]
}
```

I used `basename($mp->getDeviceFile())` to compare only block devices names and exclude the full path. 

The goal was to tell definitley the root device no matter how deep it's buried inside an lvm group or no matter what its name was as long as it's associated with / mount point it should be considered a root device.  

I thought of adding a label to root device instead. And look for it in `getRootDeviceFile`. I did not prefer adding a label because there was no way to prevent other devices from being labeled with the same label name.

Reply:
`static $blockParentDevice = NULL` 
 Prevents $blockParentDevice variable state from being overwritten or reset to NULL by instantiation due to recursion.
  
`$blockParentDevice = $items[0];`   <--- this should happens only once. It's not possible to have different devices to be mounted at mount point  /